src/common/template.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import fs from "fs-extra";
+import Handlebars from "handlebars";
+
+/**
+ * Holds a template and renders the template with the given data
+ */
+export class Template {
+  content: string;
+
+  /**
+   * Creates a Template object
+   * @param path - Path to the template file
+   * @param handlebars - Optional handlebars instance, default instance is used if not provided
+   */
+  constructor(public path: string, public handlebars = Handlebars) {
+    try {
+      this.content = fs.readFileSync(path, "utf8");
+    } catch (error) {
+      error.message = `Error initializing template: ${error.message}`;
+      throw error;
+    }
+    if (!this.content) {
+      throw new Error(`Invalid template content: ${this.content}`);
+    }
+  }
+
+  /**
+   * Make substitutions and write the template to a file.
+   *
+   * @param context - The data model to use for substitutions
+   * @param destination - The file path to write the rendered template to
+   */
+  render(context: { [key: string]: unknown }, destination: string): void {
+    fs.outputFileSync(
+      destination,
+      //Parts of the AMF model use prototype properties and methods, we need to make those available to Handlebars
+      this.handlebars.compile(this.content)(context, {
+        allowProtoPropertiesByDefault: true,
```

Comment:
Should we expose this configuration?

Reply:
probably, we can expose both compile and runtime options

Reply:
Can you think of a scenario where someone would need to worry about malicious templates?

Reply:
My personal preference is that this be minimal configuration and if someone needs something more advanced to let them fallback to using Handlebars directly.

Reply:
Agree, as the current implementation of Template is just a wrapper. If Template is going to have more logic/features (just saying, not sure it will) in future then probably it make sense to expose these options